colossalai/pipeline/rpc/_pipeline_base.py

```diff
@@ -251,7 +249,6 @@ def _make_args_kwargs(self, microbatch, merge=False):
     # TODO(jiangziyue) Consider whether this function should be protected by Lock in DAG env.
     # TODO(jiangziyue) Define a Class for DAG.
     def set_input(self, microbatch_id: int, microbatch: Tuple[Any], forward_only: bool):
-        assert self.consumer_stage_ids is not None
```

Comment:
Can you give me an example to show why this assertion is unreasonable?

Reply:
It's reasonable in previous design. But I remove the code dependent on self.consumer_stage_ids because it will introduce an extra def-use case when our design consider the situation of multithreads. In fact, it can be extracted from the topo attribute anytime we want to use it. So this dependence is not necessary.

Reply:
And also, it's main cause of data race corruption.